test-data/unit/check-generic-subtyping.test

```diff
@@ -1033,3 +1033,29 @@ x2: X2[str, int]
 reveal_type(iter(x2))  # N: Revealed type is "typing.Iterator[builtins.int*]"
 reveal_type([*x2])  # N: Revealed type is "builtins.list[builtins.int*]"
 [builtins fixtures/dict.pyi]
+
+[case testTypeSubtypingWithDifferentGenericVars]
+# flags: --python-version 3.10
+from typing import Any, Protocol
+
+class Hashable(Protocol):
+    def __hash__(self) -> int: pass
+
+t1: type = type
+h1: Hashable = t1
+
+t2: type[type] = type
+h2: Hashable = t2
+
+t3: type[Any] = type
+h3: Hashable = t3
+
+class NotHashable:
+    __hash__: None
+
+t4: type[NotHashable] = NotHashable
+h4: Hashable = t4  # E: Incompatible types in assignment (expression has type "Type[NotHashable]", variable has type "Hashable")
```

Comment:
This seems wrong. Instance of NotHashable are not hashable, but the type object itself still is. However, you can create an unhashable class by setting `__hash__ = None` on the metaclass.

Reply:
Good catch! Thanks! 👍